### PR TITLE
Update boringssl to a0fb951d2a26a8ee746b52f3ba81ab011a0af778

### PR DESCRIPTION
**Pull request by yongtang**
This fix update boringssl to `a0fb951d2a26a8ee746b52f3ba81ab011a0af778`, which contains bug fix related to #13733.

See https://github.com/tensorflow/tensorflow/pull/13734#issuecomment-337440239 for update details.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

### CODE REVIEW

**Comment by vrv** [comment]
@tensorflow-jenkins test this please

File: third_party/boringssl/add_boringssl_s390x.patch [hunk 1]
```diff
@@ -1,17 +1,6 @@
...
-     "-fno-common",
-+    "-Wno-uninitialized",
- 
-     # Modern build environments should be able to set this to use atomic
-     # operations for reference counting rather than locks. However, it's
 diff -ur a/src/include/openssl/base.h b/src/include/openssl/base.h
---- a/src/include/openssl/base.h	2017-10-10 15:50:34.000000000 +0000
-+++ b/src/include/openssl/base.h	2017-10-15 19:49:38.182154627 +0000
+--- a/src/include/openssl/base.h	2017-10-17 21:30:48.000000000 +0000
++++ b/src/include/openssl/base.h	2017-10-18 04:07:23.590036926 +0000
```
Comment (davidben) [inline third_party/boringssl/add_boringssl_s390x.patch]: Existing issue, but per the discussion here, you probably don't want to be patching this at all anymore.
https://github.com/tensorflow/tensorflow/pull/11170#issuecomment-318697044

**Comment by vrv** [comment]
@tensorflow-jenkins test this please

File: third_party/boringssl/add_boringssl_s390x.patch (hunk 1 above)
Comment (yongtang) [reply third_party/boringssl/add_boringssl_s390x.patch]: Thanks @davidben for pointing out. The patch has been removed.

**Comment by vrv** [comment]
@davidben let us know if this is okay to merge

**Comment by davidben** [comment]
(I don't actually work on Tensorflow.)

**Comment by vrv** [comment]
Yes, but I'm not sure if what your suggestion about removing the patch is what was done; I thought the idea was to remove the patch entirely, not to just undo the changes to the existing patch.

**Comment by yongtang** [comment]
@vrv In the PR the `third_party/boringssl/add_boringssl_s390x.patch` file has been deleted.

In `tensorflow/workspace.bzl` file, the boringssl section has been changed from `patched_http_archive` -> `native.http_archive` so patching is gone.

Also, in case `s390x` support is desired in TensorFlow in the future, from the comment link provided (https://github.com/tensorflow/tensorflow/pull/11170#issuecomment-329252942) it is possible to disable the GCP support to handle the big-endian issue. I don't have any access to `s390x` so I might not be able to help in that case though.

**Comment by vrv** [comment]
Ok great, thanks.  I didn't have the context.  This PR looks good, though I guess the patch file should just be deleted entirely.

**Pull request merged**